cmd/sops/main.go

```diff
@@ -443,6 +444,55 @@ func main() {
 				return nil
 			},
 		},
+		{
+			Name:      "filestatus",
+			Usage:     "check the status of the file, returning encryption status",
+			ArgsUsage: `file`,
+			Flags:     []cli.Flag{},
+			Action: func(c *cli.Context) error {
+				// _, err := config.FindConfigFile(".")
+				// if err != nil {
+				// 	return common.NewExitError(err, codes.ErrorGeneric)
+				// }
+				if c.NArg() < 1 {
+					return common.NewExitError("Error: no file specified", codes.NoFileSpecified)
+				}
+
+				fileName := c.Args()[0]
+
+				inputStore := inputStore(c, fileName)
+				outputStore := outputStore(c, fileName)
+
+				svcs := keyservices(c)
+				opts := encryptOpts{
+					Cipher:            aes.NewCipher(),
+					InputStore:        inputStore,
+					OutputStore:       outputStore,
+					InputPath:         fileName,
+					UnencryptedSuffix: "",
+					EncryptedSuffix:   "",
+					EncryptedRegex:    "",
+					KeyGroups:         []sops.KeyGroup{},
+					GroupThreshold:    0,
+					KeyServices:       svcs,
+				}
+
+				fileBytes, err := ioutil.ReadFile(opts.InputPath)
+				if err != nil {
+					return common.NewExitError(fmt.Sprintf("Error reading file: %s", err), codes.CouldNotReadInputFile)
+				}
+				branches, err := opts.InputStore.LoadPlainFile(fileBytes)
+				if err != nil {
+					return common.NewExitError(fmt.Sprintf("Error unmarshalling file: %s", err), codes.CouldNotReadInputFile)
+				}
+				if err := ensureNoMetadata(opts, branches[0]); err != nil {
+					return common.NewExitError("File is encrypted", codes.FileAlreadyEncrypted)
+				}
+
+				fmt.Println("File is unencrypted")
```

Comment:
Is `fmt` the proper way to output such an informative message?

Reply:
Looks fine to me. I'm not sure whether we should use a machine-parseable and extensible format (probably JSON), though. It would let us put more status information there in the future, e.g. whether the MAC is okay.

Reply:
Would something like `{"encrypted": true|false}` be ok?

Reply:
Looks good to me.